extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -659,108 +408,21 @@ private String getChannelTypeId(String channelId) {
         // Request new weather data to the Weather Underground service
         WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
-        weatherData = getWeatherData(USUAL_FEATURES, StringUtils.trimToEmpty(config.location));
-        return weatherData != null;
-    }
-
-    /**
-     * Request new weather data to the Weather Underground service
-     *
-     * @param features the list of features to be requested
-     * @return the weather data object mapping the JSON response or null in case of error
-     */
-    private WeatherUndergroundJsonData getWeatherData(Set<String> features, String location) {
-        WeatherUndergroundJsonData result = null;
-        boolean resultOk = false;
-        String error = null;
-        String errorDetail = null;
-        String statusDescr = null;
-
-        try {
-            // Build a valid URL for the Weather Underground service using
-            // the requested features and the thing configuration settings
-            WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
-
-            String urlStr = URL.replace("%APIKEY%", StringUtils.trimToEmpty(config.apikey));
-
-            urlStr = urlStr.replace("%FEATURES%", String.join("/", features));
-
-            String lang = StringUtils.trimToEmpty(config.language);
-            if (lang.isEmpty()) {
-                // If language is not set in the configuration, you try deducing it from the system language
-                lang = LANG_ISO_TO_WU_CODES.get(localeProvider.getLocale().getLanguage().toUpperCase());
-                logger.debug("Use language deduced from system locale {}: {}", localeProvider.getLocale().getLanguage(),
-                        lang);
-            }
-            if (lang == null || lang.isEmpty()) {
-                urlStr = urlStr.replace("%SETTINGS%", "");
-            } else {
-                urlStr = urlStr.replace("%SETTINGS%", "lang:" + lang.toUpperCase());
-            }
-
-            urlStr = urlStr.replace("%QUERY%", location);
-            logger.debug("URL = {}", urlStr);
-
-            // Run the HTTP request and get the JSON response from Weather Underground
-            String response = null;
-            try {
-                response = HttpUtil.executeUrl("GET", urlStr, FETCH_TIMEOUT_MS);
-                logger.debug("weatherData = {}", response);
-            } catch (IllegalArgumentException e) {
-                // catch Illegal character in path at index XX: http://api.wunderground.com/...
-                error = "Error creating URI with location parameter: '" + location + "'";
-                errorDetail = e.getMessage();
-                statusDescr = "@text/offline.uri-error";
-            }
+        WeatherUndergroundBridgeHandler handler = null;
+        if (this.getBridge() != null) {
+            handler = (WeatherUndergroundBridgeHandler) this.getBridge().getHandler();
```

Comment:
the `potential NPE` warning reported here can only be circumvented by assigning the bridge to a locale variable (like for the handler).